### PR TITLE
feat: github imports are now in hmi-server

### PR DESCRIPTION
**Pull request by shawnyama**
# Description

Migrated fetch requests that were done in `github-import.ts` to the hmi-server.

New proxies for:
Github API - for getting repository content/attributes
JsDelivr CDN - for downloading github files

### CODE REVIEW

File: packages/client/hmi-client/src/components/widgets/import-code-button.vue [hunk 1]
```diff
@@ -40,29 +40,36 @@ import Button from 'primevue/button';
...
+const repoOwnerAndName = ref('');
 const currentDirectory = ref('');
 const directoryContents = ref();
 const isModalVisible = ref(false);
 
 const isInDirectory = computed(() => !isEmpty(currentDirectory.value));
 
-async function initializeCodeBrowser() {
+async function getGithubRepositoryContent() {
+	const response = await API.get('/code/get_repo_content', {
```
Comment (echl) [inline packages/client/hmi-client/src/components/widgets/import-code-button.vue:59]: Can you move all API calls into a service file? See `model.ts` as an example

File: packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/code/CodeResource.java [hunk 2]
```diff
@@ -92,4 +101,21 @@ public Object findTextVars(@QueryParam("text") final String text){
 			.build();
 	}
 
+	@GET
+	@Path("/get_repo_content")
```
Comment (YohannParis) [inline packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/code/CodeResource.java:105]:
```suggestion
	@Path("/repo_content")
```

File: packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/code/CodeResource.java [hunk 3]
```diff
@@ -92,4 +101,21 @@ public Object findTextVars(@QueryParam("text") final String text){
...
+	public Response getGithubRepositoryContent(
+		@QueryParam("repoOwnerAndName") final String repoOwnerAndName,
+		@QueryParam("path") final String path
+	) {
+		return githubProxy.getGithubRepositoryContent(repoOwnerAndName, path);
+	}
+
+	@GET
+	@Path("/get_code")
```
Comment (YohannParis) [inline packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/code/CodeResource.java:114]:
```suggestion
	@Path("/repo_file_content")
```

File: packages/client/hmi-client/src/services/github-import.ts
Comment (YohannParis) [inline packages/client/hmi-client/src/services/github-import.ts:1]: Add some `try ... catch` you can see some example in ProjectService

**Review by YohannParis — APPROVED** [review]

**Pull request merged**